### ISSUE TITLE
Fail gracefully on missing yaml

### ISSUE THREAD

**Issue by wl-stepp** [pymmcore-plus/useq-schema#193]
https://github.com/pymmcore-plus/useq-schema/blob/772f59a45ea5c99299d2b2037e82de9add2607b3/src/useq/_base_model.py#L91C11-L91C12

Could this fail gracefully and suggest saving in a different format if pyyaml is not installed?
I tried to save an MDAWidget setting without having pyyaml installed. Crashed and I had to set the whole sequence up again. A one-time issue, but potentially annoying.

**Comment by tlambert03** [comment]
Yeah, that's annoying, let's remove the exception and give a warning.  Or just depend on pyyaml?

**Comment by wl-stepp** [comment]
I liked the warning for vispy in the ImageViewer widget, so I would go with that. I can see that some people would only using json...

**Comment by tlambert03** [comment]
as I implement this, I'm wondering a couple things:

1. perhaps this try/except should be done on the widget side of things.  If we do it here, we have to essentially return nonsense data to a function caller that may or may not be a widget.  It really does feel like an exception is the more "correct" thing to do (and the widget crash is a downstream concern)
2.  When you say "I tried to save an MDAWidget setting without having pyyaml installed. Crashed and I had to set the whole sequence up again." ... what context exactly were you in?  Was this your own custom GUI where you were using the MDAWidget?  I ask because "crashing" during a widget exception is something that the top level Qt-app creator should be avoiding by overriding sys.excepthook.  So if you're using this in your own app/gui, I would suggest adding logic I like I suggested to @fdrgsp  in https://github.com/fdrgsp/micromanager-gui/pull/16 (which was also related to pyyaml) and @gselzer in https://github.com/pymmcore-plus/pymmcore-widgets/issues/333#issuecomment-2231395160

**Comment by wl-stepp** [comment]
Yes, in our own GUI.
That sounds like a better fix. Thanks for pointing those out, will test.

**Comment by tlambert03** [comment]
the crashing in Qt thing is a very frequently encountered issue clearly :). yet another reason for us to try to share a unified gui
